apps/rush-lib/src/cli/test/rush-fake-build-cache-plugin/index.js

```diff
@@ -0,0 +1,22 @@
+const rushBuildCacheProvider = {
+  isCacheWriteAllowed: false,
+
+  tryGetCacheEntryBufferByIdAsync: jest.fn(),
+  trySetCacheEntryBufferAsync: jest.fn(),
+  updateCachedCredentialAsync: jest.fn(),
+  updateCachedCredentialInteractiveAsync: jest.fn(),
+  deleteCachedCredentialsAsync: jest.fn()
+};
+
+class RushFakeBuildCachePlugin {
+  static provider = rushBuildCacheProvider;
+  apply(rushSession) {
+    rushSession.hooks.initialize.tap('rush-fake-build-cache-plugin', function () {
```

Comment:
If plugin registration is based on feature kind, the plugin shouldn't be able to access any APIs of the executing Rush session until it makes a call to Rush to hand it `FeatureKinds.buildCacheProvider` and the name of the plugin, upon which Rush should hand back the `BuildCacheProviderHooks` object that contains exactly the API surface associated with being a "build cache provider".